src/middlewared/middlewared/plugins/ups.py

```diff
@@ -98,6 +98,15 @@ def driver_choices(self):
                     'Use alphanumeric characters, ".", "-" and "_"'
                 )
 
+        for field in filter(
+            lambda f: not data[f],
+            ['monpwd', 'monuser', 'identifier', 'shutdowncmd']
+        ):
+            verrors.add(
```

Comment:
Why didn't you just mark the field as required?

Reply:
@themylogin because this is a config service and we have `update=True` here. Also required doesn't check for empty string iirc

Reply:
`empty=False` does

Reply:
This simplifies a lot of things, will update the PR. Thank you